memo_app.rb

```diff
@@ -53,48 +64,50 @@ def find_memo(id)
 end
 
 post '/create' do
-  add_memo(params)
-  redirect '/'
+  if params['title'].empty?
+    show_error_message
+    erb :new
+  else
+    add_memo(params)
+    redirect '/'
+  end
 end
 
 get '/:id' do
-  @memo = find_memo(params['id'])
-  if @memo.nil?
-    status 404
-  else
-    erb :show
-  end
+  @memo = search_memo(params['id'])
+  status 404 if @memo.nil?
+  erb :show
```

Comment:
このコードでも動くかもしれませんが、`@memo`がnilのときは`status 404`と`erb :show`の両方が呼ばれるように見えてややこしいので、どちらか一方だけを呼ぶ形式にした方がいいと思います。

```suggestion
  if @memo
    erb :show
  else
    status 404
  end
```

なお、上の例では正常系が先に、異常系があとになるようなコードにしています。
（通常の操作なら正常系に入ることが大半だと思うので）

Reply:
私もこちらの方がわかりやすくて好きです。
好まれるコードを沢山読み、自分の感覚に落とし込みつつ、引き出せるようにします。